src/uu/slabtop/src/parse.rs

```diff
@@ -0,0 +1,301 @@
+            // <num_slabs>
+            'l' => sort("num_slabs"),
+            // <active_slabs>
+            'v' => sort("active_slabs"),
+            // name, sort by lexicographical order
+            'n' => self.data.sort_by(|(name1, _), (name2, _)| {
+                if ascending_order {
+                    name1.cmp(name2)
+                } else {
+                    name2.cmp(name1)
+                }
+            }),
+            // <pagesperslab>
+            'p' => sort("pagesperslab"),
+            // <objsize>
+            's' => sort("objsize"),
+            // sort by cache utilization
+            'u' => {
+                let offset_active_objs = self.offset("active_objs");
+                let offset_num_objs = self.offset("num_objs");
+
+                if let (Some(offset_active_objs), Some(offset_num_objs)) =
+                    (offset_active_objs, offset_num_objs)
+                {
+                    self.data.sort_by(|(_, data1), (_, data2)| {
+                        let cu = |active_jobs: Option<&u64>, num_jobs: Option<&u64>| match (
+                            active_jobs,
+                            num_jobs,
+                        ) {
+                            (Some(active_jobs), Some(num_jobs)) => {
+                                Some((*active_jobs as f64) / (*num_jobs as f64))
+                            }
+                            _ => None,
+                        };
+                        let cu1 = cu(data1.get(offset_active_objs), data1.get(offset_num_objs));
+                        let cu2 = cu(data2.get(offset_active_objs), data2.get(offset_num_objs));
+
+                        if let (Some(cu1), Some(cu2)) = (cu1, cu2) {
+                            let result = if ascending_order {
+                                cu1.partial_cmp(&cu2)
+                            } else {
+                                cu2.partial_cmp(&cu1)
+                            };
+                            match result {
+                                Some(ord) => ord,
+                                None => Ordering::Equal,
+                            }
+                        } else {
+                            Ordering::Equal
+                        }
+                    })
+                }
+            }
+
+            // <num_objs>
+            // Default branch : `o`
+            _ => sort("num_objs"),
+        }
+
+        self
+    }
+
+    fn offset(&self, meta: &str) -> Option<usize> {
+        self.meta.iter().position(|it| it.eq(meta))
+    }
+
+    /////////////////////////////////// helpers ///////////////////////////////////
+
+    #[inline]
+    fn total(&self, meta: &str) -> u64 {
+        let Some(offset) = self.offset(meta) else {
+            return 0;
+        };
+
+        self.data
+            .iter()
+            .filter_map(|(_, data)| data.get(offset))
+            .sum::<u64>()
+    }
+
+    pub(crate) fn object_minimum(&self) -> u64 {
```

Comment:
why pub(crate)?

Reply:
It used [here](https://github.com/uutils/procps/pull/42/files/152224f20adcee0c151ca10d69a24aa7c4580975#diff-56a5097cae513435b8fee1c4284e5ca02d0f74ef86aed893b59a9188e672f294R55-R60),
```rust
    println!(
        r" Minimum / Average / Maximum Object : {:.2}K / {:.2}K / {:.2}K",
        to_kb(slabinfo.object_minimum()),
        to_kb(slabinfo.object_avg()),
        to_kb(slabinfo.object_maximum())
    );
```
 I think using pub(crate) helps with API management?